lib/multi.c

```diff
@@ -2019,6 +2019,19 @@ static CURLMcode multi_runsingle(struct Curl_multi *multi,
       break;
 
     case MSTATE_DO:
+      if(data->set.fprereq) {
+        /* call the prerequest callback function */
+        Curl_set_in_callback(data, true);
+        data->set.fprereq(data->set.prereq_userp,
+                          data->info.conn_primary_ip,
+                          data->info.conn_local_ip,
+                          data->info.conn_scheme,
+                          data->info.conn_protocol,
+                          data->info.conn_primary_port,
+                          data->info.conn_local_port);
```

Comment:
Should we consider allowing `curl_easy_getinfo()` calls from within this callback to retrieve all that information instead of passing these six specific arguments?

Reply:
Interesting. Maybe? That's a departure from all existing callbacks AFAIK and I don't know if accessing `curl_easy_getinfo()` is supported during callbacks during a transfer - I thought it was only a post-transfer thing. This is going off vague memories from IRC though.

Reply:
@bagder is there anything else in the code that uses `curl_easy_getinfo` in flight? I'd have thought there was a danger of some variables having not been populated.